src/HIE/Bios/Cradle.hs

```diff
@@ -483,69 +579,109 @@ processCabalWrapperArgs args =
 -- arguments to the executable.
 type GhcProc = (FilePath, [String])
 
--- | Generate a fake GHC that can be passed to cabal
+-- | Generate a fake GHC that can be passed to cabal or stack
 -- when run with --interactive, it will print out its
 -- command-line arguments and exit
-withCabalWrapperTool :: GhcProc -> FilePath -> IO FilePath
-withCabalWrapperTool (mbGhc, ghcArgs) wdir = do
-    cacheDir <- getCacheDir ""
-    createDirectoryIfMissing True cacheDir
+withGhcWrapperTool :: GhcProc -> FilePath -> IO FilePath
+withGhcWrapperTool (mbGhc, ghcArgs) wdir = do
     let wrapperContents = if isWindows then cabalWrapperHs else cabalWrapper
-        suffix fp = if isWindows then fp <.> "exe" else fp
-    let srcHash = show (fingerprintString wrapperContents)
-    let wrapper_name = "wrapper-" ++ srcHash
-    let wrapper_fp = suffix $ cacheDir </> wrapper_name
-    exists <- doesFileExist wrapper_fp
-    unless exists $
+        withExtension fp = if isWindows then fp <.> "exe" else fp
+        srcHash = show (fingerprintString wrapperContents)
+    cacheFile (withExtension "wrapper") srcHash $ \wrapper_fp ->
       if isWindows
-      then do
+      then
         withSystemTempDirectory "hie-bios" $ \ tmpDir -> do
-          let wrapper_hs = cacheDir </> wrapper_name <.> "hs"
+          let wrapper_hs = wrapper_fp -<.> "hs"
           writeFile wrapper_hs wrapperContents
           let ghc = (proc mbGhc $
                       ghcArgs ++ ["-rtsopts=ignore", "-outputdir", tmpDir, "-o", wrapper_fp, wrapper_hs])
                       { cwd = Just wdir }
           readCreateProcess ghc "" >>= putStr
```

Comment:
What is this for, btw?

Reply:
We basically need a shim for ghc on windows to intercept the ghc arguments cabal would pass to ghc. However, there is no simple shim available, thus we compile a hs file that is the shim.